### PR TITLE
integrations/operator: Add AccessList support

### PR DESCRIPTION
**Pull request by hugoShaka**
Changelog: Adds AcessList support to the Teleport Operator.

This PR adds AccessList support to the operator.

As this is an enterprise feature, the tests are running against teleport.e (a PR will follow).

I faced 2 major issues when implementing ALs support:
- we broke the naming convention between proto and go structs, I changed `crdgen` to default to the proto name instead of the jsonName (this is backward compatible because `json_tag` is set everywhere else, the snapshot tests are still passing)
- proto and go structs do not share the same structure anymore, I added an override on the Traits field.

### CODE REVIEW

File: integrations/operator/controllers/resources/accesslist_controller.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+func (r accessListClient) Create(ctx context.Context, accessList *accesslist.AccessList) error {
+	teleportClient, err := r.TeleportClientAccessor(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	_, err = teleportClient.AccessListClient().UpsertAccessList(ctx, accessList)
+	return trace.Wrap(err)
+}
+
+// Update updates a Teleport github_connector
```
Comment (mdwn) [inline integrations/operator/controllers/resources/accesslist_controller.go:40]:
```suggestion
// Update updates a Teleport access_list
```

File: integrations/operator/controllers/resources/accesslist_controller.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	teleportClient, err := r.TeleportClientAccessor(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	_, err = teleportClient.AccessListClient().UpsertAccessList(ctx, accessList)
+	return trace.Wrap(err)
+}
+
+// Delete deletes a Teleport github_connector
```
Comment (mdwn) [inline integrations/operator/controllers/resources/accesslist_controller.go:51]:
```suggestion
// Delete deletes a Teleport access_list
```

File: integrations/operator/controllers/resources/accesslist_controller.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+func (r accessListClient) Delete(ctx context.Context, name string) error {
+	teleportClient, err := r.TeleportClientAccessor(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return trace.Wrap(teleportClient.AccessListClient().DeleteAccessList(ctx, name))
+}
+
+// NewAccessListReconciler instantiates a new Kubernetes controller reconciling github_connector resources
```
Comment (mdwn) [inline integrations/operator/controllers/resources/accesslist_controller.go:61]:
```suggestion
// NewAccessListReconciler instantiates a new Kubernetes controller reconciling access_list resources
```

File: integrations/operator/controllers/resources/accesslist_controller.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	resourcesv1 "github.com/gravitational/teleport/integrations/operator/apis/resources/v1"
+	"github.com/gravitational/teleport/integrations/operator/sidecar"
+)
+
+// accessListClient implements TeleportResourceClient and offers CRUD methods needed to reconcile access_lists.
+type accessListClient struct {
+	TeleportClientAccessor sidecar.ClientAccessor
+}
+
+// Get gets the Teleport github_connector of a given name
```
Comment (mdwn) [inline integrations/operator/controllers/resources/accesslist_controller.go:19]:
```suggestion
// Get gets the Teleport access_list of a given name
```

**Review by mdwn — APPROVED** [review]
So IIRC, you could potentially excise some of the unrelated changes here, e.g. `userloginstate` and whatnot. However, I'd rather we get these in so that we're not required to do manual wrangling every time we update the operator. LGTM!

File: integrations/operator/crdgen/schemagen.go [hunk 5]
```diff
@@ -136,7 +136,10 @@ func (generator *SchemaGenerator) addResource(file *File, name string, opts ...r
 	} else {
 		specField, ok := rootMsg.GetField("Spec")
 		if !ok {
-			return trace.NotFound("message %q does not have Spec field", name)
+			specField, ok = rootMsg.GetField("spec")
+			if !ok {
+				return trace.NotFound("message %q does not have Spec field", name)
+			}
```
Comment (marcoandredinis) [inline integrations/operator/crdgen/schemagen.go:142]: Isn't it the same thing? :thinking:

File: integrations/operator/apis/resources/v1/accesslist_types.go [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+				Description: l.Annotations[resources.DescriptionKey],
+				Labels:      l.Labels,
+			},
+		},
+		Spec: accesslist.Spec(l.Spec),
+	}
+	return resource
+}
+
+func (l *TeleportAccessList) StatusConditions() *[]metav1.Condition {
```
Comment (marcoandredinis) [inline integrations/operator/apis/resources/v1/accesslist_types.go:84]: Missing godoc

**Review by marcoandredinis — APPROVED** [review]

File: integrations/operator/crdgen/schemagen.go (hunk 5 above)
Comment (mdwn) [reply integrations/operator/crdgen/schemagen.go:142]: Spec with a capital S vs. spec with a lowercase S.  :-)

Comment (marcoandredinis) [reply integrations/operator/crdgen/schemagen.go:142]: 🤯 
That is very subtle
Can we add a comment?

**Review by r0mant — APPROVED** [review]
Bot.

**Review by tigrato — APPROVED** [review]

**Comment by hugoShaka** [comment]
Blocked by https://github.com/gravitational/teleport/issues/35088

**Pull request merged**